### ISSUE TITLE
Empty lambdas are sometimes considered definitionally equal, other times not

### ISSUE THREAD

**Issue by identicalsnowflake** [agda/agda#3176]
```agda
open import Agda.Builtin.Nat renaming (Nat to ℕ)
open import Agda.Builtin.Equality

record Σ (A : Set) (B : A → Set) : Set where
  constructor _,_
  field
    proj₁ : A
    proj₂ : B proj₁

open Σ public

∃ : ∀ {A : Set} → (A → Set) → Set
∃ = Σ _

_×_ : ∀ _ _ → _
_×_ a b = ∃ λ (_ : a) → b

record Fin (n : ℕ) : Set where
  field
    m : ℕ
    .p : ∃ λ k → (ℕ.suc k + m) ≡ n


module _ (n : ℕ) where

  data S : Set where
    V : S
    K : Fin n → S
    _X_ : S → S → S

[_] : ∀ {n} → S n → (Fin n → Set) → Set → Set
[ V ] k t = t
[ K i ] k _ = k i
[ s₁ X s₂ ] k t = [ s₁ ] k t × [ s₂ ] k t

postulate
  ignore : ∀ {t : Set} → t

∅ : Fin 0 → Set
∅ ()

_<*>_ : ∀ {s : S 0} {t₁ t₂}
  → [ s ] (λ ()) (t₁ → t₂)
  → [ s ] (λ ()) t₁
  → [ s ] (λ ()) t₂
_<*>_ {s₁ X s₂} {t₁} {t₂} (f , _) (x , _) =
  let v : [ s₁ ] (λ ()) t₂
      v = _<*>_ {s = s₁} {t₁ = t₁} {t₂ = t₂} {!!} {!!}
  in
  ignore
_<*>_ f x = ignore
```

The above should load into Agda. If you put the cursor in the first hole and try to provide the solution `f`, Agda will accept this solution. But if you reload the file, Agda will then reject that solution (with a strange error that doesn't seem related to the problem, actually -- it says `(s₁ X s₂) != s₁ of type (S 0)`).

However, if you replace the 4 occurrences of `(λ ())` with `∅`, the solution is accepted and everything works as expected.

Anyway, the fact that the declaration `v` in the original form is even accepted at all means that Agda considered `(λ ())` equal to `(λ ())`, so it's weird that it considers them equal in that case, it accepts this equality again when the solution `f` is provided in the first hole, but then decides they aren't equal (with a seemingly unrelated error) when you reload.

**Comment by andreasabel** [comment]
Mmh, the current check for equality of absurd lambdas was weird:  https://github.com/agda/agda/blob/2796d596ffc458821fe2f1aa88d0c870278201a9/src/full/Agda/TypeChecking/Conversion.hs#L540-L549
If the absurd lambdas were internally represented as _different functions_, they were treated _equal_, but if they were the _same_ function, they were only treated equal if they were applied to equal arguments.  WTF?

I now changed it to have: _all absurd lambdas are created equal_.

I think this was the intention behind the fix for #857.

**Comment by andreasabel** [comment]
Thanks for reporting!
